numba_rvsdg/tests/simulator.py

```diff
@@ -427,3 +435,11 @@ def op_POP_JUMP_FORWARD_IF_NONE(self, inst):
     def op_POP_JUMP_BACKWARD_IF_NONE(self, inst):
         self.branch = self.stack[-1] is None
         self.stack.pop()
+
+    def op_END_FOR(self, inst):
+        self.stack.pop()
+        self.stack.pop()
```

Comment:
Since `END_FOR` was introduced with 3.12, it'll need a `PYVERSION` guard too, in the same style as above for `FOR_ITER`.